ksqldb-test-util/src/main/java/io/confluent/ksql/test/util/KsqlTestFolder.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.test.util;
+
+import java.io.File;
+import java.lang.ref.WeakReference;
+import org.junit.rules.TemporaryFolder;
+
+/**
+ * A Utility class to consolidate where we allocate state folders for tests.
+ */
+public final class KsqlTestFolder {
+
+  /**
+   * Uninstantiable static utility class
+   */
+  private KsqlTestFolder() { }
+
+
+  /**
+   * Create a temporary folder for testing. Rather than the system default
+   * temp directory, use the current working directory to avoid directories
+   * being deleted during tests.
+   */
+  public static TemporaryFolder temporaryFolder() {
+    final TemporaryFolder temporaryFolder = TemporaryFolder
+        .builder()
+        .assureDeletion()
+        .parentFolder(new File(System.getProperty("user.dir")))
```

Comment:
By the way, this unintuitively named system property resolves to the current working directory.

Reply:
It's just part of the JVM: https://docs.oracle.com/javase/tutorial/essential/environment/sysprop.html

Reply:
Ah, I see what you mean. No; it'd be nice if they had an enum or something, but it's just a string.